pg2kinesis/formatter.py

```diff
@@ -55,46 +64,102 @@ def _preprocess_change(self, change):
                 try:
                     mat = self._primary_key_patterns[rec[1]].search(rec[3])
                 except KeyError:
-                    self._log_and_raise('Unable to locate table: "{}"'.format(rec[1]))
+                    self._log_and_raise(MISSING_TABLE_ERR.format(rec[1]))
                 else:
                     if mat:
                         pkey = mat.groups()[0]
-                        return Change(xid=self.cur_xact, table=table_name, operation=rec[2][:-1], pkey=pkey)
+                        return [Change(xid=self.cur_xact, table=table_name,
+                                       operation=rec[2][:-1], pkey=pkey)]
                     else:
-                        # TODO: make this an error or warning.
-                        # self._log_and_raise('Unable to locate primary key for table "{}"'.format(table_name))
-                        pass
+                        self._log_and_raise(MISSING_PK_ERR.format(table_name))
         else:
             self._log_and_raise('Unknown change: "{}"'.format(change))
 
+        return []
+
+    def _preprocess_wal2json_change(self, change):
+        """
+        Takes a message payload from the wal2json plugin and distills it into a
+        list of Change or FullChange tuples.
+
+        They look like this:
+            {
+                "xid": 1234567890
+                "change": [
+                    {
+                        "kind": "insert",
+                        "schema": "public",
+                        "table": "some_table",
+                        "columnnames": ["id"],
+                        "columntypes": ["int4"],
+                        "columnvalues": [42]
+                    }
+                ]
+            }
+        :param change: a message payload from postgres wal2json plugin.
+        :return: A list of type Change or FullChange
+        """
+
+        change_dictionary = json.loads(change)
+        if not change_dictionary:
+            return None
+
+        self.cur_xact = change_dictionary.get('xid')
```

Comment:
nit. For keys that should always be in the dict I think you should use __itemgetter__  i.e.`change_dictionary['xid']`